dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -76,11 +76,18 @@ int64_t seek_memory_video_file(void *data_ptr, int64_t new_position, int origin)
 
 using AVPacketScope = std::unique_ptr<AVPacket, decltype(&av_packet_unref)>;
 
-const std::vector<AVCodecID> FramesDecoder::SupportedCodecs = {
+span<const AVCodecID> FramesDecoder::SupportedCodecs() const {
+  static constexpr std::array<AVCodecID, 7> codecs = {
     AVCodecID::AV_CODEC_ID_H264,
     AVCodecID::AV_CODEC_ID_HEVC,
-    AVCodecID::AV_CODEC_ID_MPEG4
-};
+    // TODO(janton): AVCodecID::AV_CODEC_ID_MPEG4, -> not supported by current FFmpeg build
+    AVCodecID::AV_CODEC_ID_VP8,
+    AVCodecID::AV_CODEC_ID_VP9,
+    // TODO(janton): AVCodecID::AV_CODEC_ID_AV1, -> not supported by most platforms
```

Comment:
For CPU, maybe we can have a separate list for GPU and CPU?

Reply:
I gave it a try, but I hit a roadblock. To be continued in a follow-up

Reply:
Sure thing. It is big enough already.